hecuba/storageobj.py

```diff
@@ -350,43 +348,56 @@ def make_persistent(self, name):
             log.error("Unable to execute %s", query_simple)
             raise ir
 
+        #Iterate over the objects the user has requested to be persistent
+        # retrieve them from memory and make them persistent
         for obj_name, obj_info in self._persistent_props.items():
-            if hasattr(self, obj_name):
-                pd = getattr(self, obj_name)
+            try:
+                pd = object.__getattribute__(self, obj_name)
                 if obj_info['type'] not in IStorage._basic_types:
                     sd_name = self._ksp + "." + self._table + "_" + obj_name
                     pd.make_persistent(sd_name)
-                setattr(self, obj_name, pd)  # super(StorageObj, self).__setattr__(obj_name, pd) why?
-
+                #self is persistent so setting the attribute will store the data and create the appropiate binding
+                setattr(self, obj_name, pd)
+            except AttributeError:
+                #Attribute unset, no action needed
+                pass
         self._store_meta(self._build_args)
 
     def stop_persistent(self):
         """
             The StorageObj stops being persistent, but keeps the information already stored in Cassandra
         """
+        for obj_name, obj_info in self._persistent_props.items():
```

Comment:
check this: why are you using build in functions? Can't you just check if they are IStorage instances?

Reply:
We only need to stop the objects which are in memory, stopping something that is in Cassandra doesn't do anything. Stop_persistent only switches from persistent to non persistent (without deleting anything).
About the second point, yes, it's the same. In this way getattribute is called fewer times if we check the type before. However, it does a search in a list which isn't efficient too. I will change it.